lib/perl/Genome/Model/Tools/Vcf/EvaluateVcf.pm

```diff
@@ -0,0 +1,790 @@
+    return $stat;
+}
+
+sub stat_false_positive_exact {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{false_positive_exact};
+    return $stat;
+}
+
+sub stat_false_positive_partial {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{false_positive_partial};
+    return $stat;
+}
+
+sub stat_true_negatives {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{true_negatives};
+    return $stat;
+}
+
+sub stat_exact_specificity {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+
+    # exact specificity, these are not strictly accurate as the tn_bed
+    # may be significantly smaller than the target space ROI
+    
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat =
+      ($results->{true_negatives} - $results->{false_positive_exact}) / $tn;
+    return $stat;
+}
+
+sub stat_partial_specificity {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat =
+      ($results->{true_negatives} - $results->{false_positive_partial}) / $tn;
+    return $stat;
+}
+
+sub stat_exact_ppv {
+    my $self    = shift;
+
+    my $fp_exact = $self->stat_false_positive_exact();
+    my $tp_exact = $self->stat_true_positive_found_exact();
+
+    my $denominator = $fp_exact + $tp_exact;
+    if ($denominator == 0) {
+        return 0;
+    }
+
+    my $stat = $tp_exact / $denominator;
+    return $stat;
+}
+
+sub stat_partial_ppv {
+    my $self    = shift;
+
+    my $fp_partial = $self->stat_false_positive_partial();
+    my $tp_partial = $self->stat_true_positive_found_partial();
+
+    my $denominator = $fp_partial + $tp_partial;
+    if ($denominator == 0) {
+        return 0;
+    }
+
+    my $stat = $tp_partial / $denominator;
+    return $stat;
+}
+
+sub stat_vcf_lines_overlapping_tn {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+    my $stat    = $results->{false_positives_in_roi};
+    return $stat;
+}
+
+sub stat_lines_specificity_in_tn_only {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+
+    # this is more accurate than the other specificity measures, but
+    # doesn't take partial into account
+
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat = ($tn - $results->{false_positives_in_roi}) / $tn;
+    return $stat;
+}
+
+sub help_brief {
+    return "Compare or Validate VCF files to a Gold Standard";
+}
+
+sub help_detail {
+    my $doc = q{
+
+    BACKGROUND
+    ==========
+
+    For validation, it is important to compare calls from the pipeline +
+    reporting framework and calculate sensitivity, specificity and positive
+    predictive value of the pipeline's calls versus the gold standard.
```

Comment:
This first sentence is a little cumbersome.  I'm a bit confused which parts go together and which are merely listed sequentially.